starks/finitefield.py

```diff
@@ -101,6 +120,136 @@ def inverse(self):
         )
 
       return Fq(x) * Fq(d.coefficients[0].inverse())
+    
+    # less than or equal operation
+    def LQ(self, other):
+      if self.poly.degree() > other.poly.degree():
+        return 0
+      elif other.poly.degree() > self.poly.degree():
+        return 1
+
+      xor_o = self + other
+
+      if xor_o == Fq(0):
+        return 1
+      
+      
+      if str(self.poly.coefficients[xor_o.poly.degree()])[0] > str(other.poly.coefficients[xor_o.poly.degree()])[0]:
+        return 0
+      else:
+        return 1
+
+    # two pow self operation
+    def pow(self):
+      num = Fq([1])
+    
+      for i in range(self.poly.degree()+1):
+        if str(self.poly.coefficients[i])[0] == '1':
+          l = num_to_binary_list_pow(i+1)
+          num = num * Fq(l)
+      
+      return num
+
+    # addition floating point operation
+    def addition(self, v_a, p_a, z_a, s_a, v_b, p_b, z_b, s_b):
```

Comment:
Let's start to think about factoring floating point numbers out into their own class. This will let us implement logistic regression and other algorithms in a way that looks mathematically natural. We can do this in a future PR